implementations/rust/kex/src/x3dh.rs

```diff
@@ -215,11 +219,12 @@ impl std::fmt::Debug for X3dhInitiator {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
         write!(
             f,
-            r#"X3dhInitiator {{ ephemeral_identity_key: {:?}, prekey_bundle: {:?}, state: {:?}, vault, completed_key_exchange: {:?} }}"#,
+            r#"X3dhInitiator {{ ephemeral_identity_key: {:?}, prekey_bundle: {:?}, state: {:?}, vault, completed_key_exchange: {:?}, identity_key: {:?} }}"#,
```

Comment:
It would be preferable to implement `Debug` using the `debug_struct` etc method, example here: https://doc.rust-lang.org/std/fmt/trait.Debug.html#examples-1

also for the Debug implementation of `X3dhResponder`

Reply:
The benefit is that using `{:?}` is printed on one line, and `{:#?}` is printed on many lines. this is more important if these formatted strings ever get logged, it becomes really hairy to untangle single line log messages from multi-line.

Reply:
I get that. The problem is you can't use `debug_struct` on Arc<Mutex<dyn DynVault>>

Reply:
I think you can pretty much use it regardless... but either way,`X3dhInitiator` is the name of the struct you'd pass to `debug_struct` though, not the `Arc`. 

The main problem I have with this is the multi-line printing... if you don't want to do the work to implement with `debug_struct`, I think at minimum you should remove the multi-line value and put it all on a single line.

Reply:
If I put in the debug_struct it doesn't compile.

Reply:
That's because there are non-Debug impl types in the struct fields. 

Just make it one single line then, to keep it consistent?

Reply:
The compiler complains that Debug is not implemented for trait objects. While it is good practice to use derive(Debug) when you can, its not always feasible.